### PR TITLE
[CodingStyle] Add DataProviderArrayItemsNewlinedRector

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

**Comment by samsonasik** [comment]
Handle new line may need handle `<=0` check tweak to avoid crash or infinite loop on multiple rules usage, see NewlineAfterStatementRector;

https://github.com/rectorphp/rector-src/blob/e8dd953f10a7afc3610d5185c6b42216e6d9e050/rules/CodingStyle/Rector/Stmt/NewlineAfterStatementRector.php#L149-L153

File: rules-tests/CodingStyle/Rector/ClassMethod/DataProviderArrayItemsNewlinedRector/Fixture/some_class.php.inc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * @dataProvider provideData()
+     */
+    public function testGetBytesSize(string $content, int $number): void
+    {
+        // ...
+    }
+
+    public function provideData(): array
+    {
+        return [
```
Comment (staabm) [inline rules-tests/CodingStyle/Rector/ClassMethod/DataProviderArrayItemsNewlinedRector/Fixture/some_class.php.inc:43]: Maybe turn them into one `yield` per array item instead?

Comment (TomasVotruba) [reply rules-tests/CodingStyle/Rector/ClassMethod/DataProviderArrayItemsNewlinedRector/Fixture/some_class.php.inc:43]: There another rule for that already :) 
This rule is mainly for Test Gen AI styling.

**Comment by TomasVotruba** [comment]
> Handle new line may need handle <=0 check tweak to avoid crash or infinite loop on multiple rules usage, see NewlineAfterStatementRector;

That could work too, thanks!
I've a handy method from standard printer that handles this by itself :) I'm happy php-parser has exact method we need here.

**Pull request merged**